src/ssl.c

```diff
@@ -3456,10 +3456,13 @@ int PemToDer(const unsigned char* buff, long longSz, int type,
 
     headerEnd += XSTRLEN(header);
 
+    if ((int)(headerEnd - (char*)buff) > sz - 2)
+        return SSL_BAD_FILE;
+
     /* eat end of line */
-    if (headerEnd[0] == '\n')
+    if (headerEnd && headerEnd[0] == '\n')
```

Comment:
If headerEnd is redundant here.  There is a !headerEnd check a few lines above making sure the pointer isn't NULL.  Adding to that pointer can't make it null.